shared/util/tf/tf_generator.go

```diff
@@ -0,0 +1,439 @@
+
+	rootBody := hclFile.Body()
+	err = tf.generateAwsProvider(rootBody, "", "${var.aws_account_id}", "${var.aws_role}")
+	if err != nil {
+		return errors.Wrap(err, "Unable to generate HCL code for AWS provider")
+	}
+
+	err = tf.generateAwsProvider(rootBody, "czi-si", "626314663667", "tfe-si")
+	if err != nil {
+		return errors.Wrap(err, "Unable to generate HCL code for AWS provider")
+	}
+
+	eksBody := rootBody.AppendNewBlock("data", []string{"aws_eks_cluster", "cluster"}).Body()
+	eksBody.SetAttributeRaw("name", tokens("var.k8s_cluster_id"))
+	eksAuthBody := rootBody.AppendNewBlock("data", []string{"aws_eks_cluster_auth", "cluster"}).Body()
+	eksAuthBody.SetAttributeRaw("name", tokens("var.k8s_cluster_id"))
+
+	kubernetesProviderBody := rootBody.AppendNewBlock("provider", []string{"kubernetes"}).Body()
+	kubernetesProviderBody.SetAttributeRaw("host", tokens("data.aws_eks_cluster.cluster.endpoint"))
+	kubernetesProviderBody.SetAttributeRaw("cluster_ca_certificate", tokens("base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)"))
+	kubernetesProviderBody.SetAttributeRaw("token", tokens("data.aws_eks_cluster_auth.cluster.token"))
+
+	kubeNamespaceBody := rootBody.AppendNewBlock("data", []string{"kubernetes_namespace", "happy-namespace"}).Body()
+	kubeNamespaceBody.AppendNewBlock("metadata", nil).Body().SetAttributeRaw("name", tokens("var.k8s_namespace"))
+
+	awsAliasTokens := tokens("aws.czi-si")
+	appKeyBody := rootBody.AppendNewBlock("data", []string{"aws_ssm_parameter", "dd_app_key"}).Body()
+	appKeyBody.SetAttributeValue("name", cty.StringVal("/shared-infra-prod-datadog/app_key"))
+	appKeyBody.SetAttributeRaw("provider", awsAliasTokens)
+	apiKeyBody := rootBody.AppendNewBlock("data", []string{"aws_ssm_parameter", "dd_api_key"}).Body()
+	apiKeyBody.SetAttributeValue("name", cty.StringVal("/shared-infra-prod-datadog/api_key"))
+	apiKeyBody.SetAttributeRaw("provider", awsAliasTokens)
+
+	datadogProviderBody := rootBody.AppendNewBlock("provider", []string{"datadog"}).Body()
+	datadogProviderBody.SetAttributeRaw("app_key", tokens("data.aws_ssm_parameter.dd_app_key.value"))
+	datadogProviderBody.SetAttributeRaw("api_key", tokens("data.aws_ssm_parameter.dd_api_key.value"))
+
+	_, err = tfFile.Write(hclFile.Bytes())
+
+	return err
+}
+
+func (tf TfGenerator) generateAwsProvider(rootBody *hclwrite.Body, alias, accountIdExpr, roleExpr string) error {
+	awsProviderBody := rootBody.AppendNewBlock("provider", []string{"aws"}).Body()
+	if alias != "" {
+		awsProviderBody.SetAttributeValue("alias", cty.StringVal(alias))
+	}
+	awsProviderBody.SetAttributeValue("region", cty.StringVal(*tf.happyConfig.AwsRegion()))
+
+	assumeRoleBlockBody := awsProviderBody.AppendNewBlock("assume_role", nil).Body()
+	assumeRoleBlockBody.SetAttributeRaw("role_arn", tokens(fmt.Sprintf("\"arn:aws:iam::%s:role/%s\"", accountIdExpr, roleExpr)))
+	awsProviderBody.SetAttributeRaw("allowed_account_ids", tokens(fmt.Sprintf("[\"%s\"]", accountIdExpr)))
```

Comment:
right now the provider.tf is a static file. I think we should use a template for it since there is no dynamic configuration.

Reply:
There is. Region is dynamic.